public/global.d.ts

```diff
@@ -23,12 +21,604 @@ interface JQuery {
     slider(method: string, func: string, options?: any): JQuery;
     cropper(options?: any): JQuery;
     izoomify(options?: any): JQuery;
+
+    //#region select2
+
+    /**
+     * Initializes or modifies a select2 instance with provided options
+     *
+     * @param options - Configuration options for the select2 instance
+     * @returns The jQuery object for chaining
+     */
+    select2(options?: Select2Options): JQuery;
+
+    /**
+     * Retrieves data currently selected in the select2 instance
+     *
+     * @param field - A string specifying the 'data' method
+     * @returns An array of selected items
+     */
+    select2(field: 'data'): any[];
+
+    /**
+     * Calls the specified select2 method
+     *
+     * @param method - The name of the select2 method to invoke
+     * @returns The jQuery object for chaining
+     */
+    select2(method: 'open' | 'close' | 'destroy' | 'focus' | 'val', value?: any): JQuery;
+
+    //#endregion
+
+    //#region sortable
+
+    /**
+     * Initializes or updates a sortable instance with the provided options
+     *
+     * @param options - Configuration options for the sortable instance
+     * @returns The jQuery object for chaining
+     */
+    sortable(options?: SortableOptions): JQuery;
+
+    /**
+     * Calls a sortable method to perform actions on the instance
+     *
+     * @param method - The name of the sortable method to invoke
+     * @returns The jQuery object for chaining
+     */
+    sortable(method: 'destroy' | 'disable' | 'enable' | 'refresh' | 'toArray'): JQuery;
+
+    /**
+     * Retrieves the sortable's instance object. If the element does not have an associated instance, undefined is returned.
+     *
+     * @returns The instance of the sortable object
+     */
+    sortable(method: 'instance'): object;
+
+    /**
+     * Retrieves the current option value for the specified option
+     *
+     * @param method - The string 'option' to retrieve an option value
+     * @param optionName - The name of the option to retrieve
+     * @returns The value of the specified option
+     */
+    sortable(method: 'option', optionName: string): any;
+
+    /**
+     * Sets the value of the specified option
+     *
+     * @param method - The string 'option' to set an option value
+     * @param optionName - The name of the option to set
+     * @param value - The value to assign to the option
+     * @returns The jQuery object for chaining
+     */
+    sortable(method: 'option', optionName: string, value: any): JQuery;
+
+    /**
+     * Sets multiple options using an object
+     *
+     * @param method - The string 'option' to set options
+     * @param options - An object containing multiple option key-value pairs
+     * @returns The jQuery object for chaining
+     */
+    sortable(method: 'option', options: SortableOptions): JQuery;
+
+    //#endregion
+}
+
+/**
+ * Toastr is a global object for displaying toast notifications
+ */
+declare var toastr: {
```

Comment:
toastr already pulled types for me even without custom declaration. Hasn't it been for you?
Also, methods return JQuery<HTMLElement>

Reply:
Nope. Did not. I could type `toastr` quite fine, but it did not auto-complete to any of the available method names that you could call on that. Even less so to what kind of parameters those have.  
That's what this PR should add.

Reply:
This is default:
![image](https://github.com/SillyTavern/SillyTavern/assets/9962104/3d62bbb2-104b-4116-8f06-4bf5441e4484)
![image](https://github.com/SillyTavern/SillyTavern/assets/9962104/b478fd09-fe36-4be1-bf30-35dcde26aeb2)

This is with this PR:
![image](https://github.com/SillyTavern/SillyTavern/assets/9962104/fc1da037-d1cb-4d4e-9201-edb49430fc91)
![image](https://github.com/SillyTavern/SillyTavern/assets/9962104/a717eb54-2605-4a3a-8876-9b473f7fb150)

Reply:
No, I don't. Perhaps the var in this file overwrites the declaration added by types pulled here: https://github.com/SillyTavern/SillyTavern/blob/release/public/jsconfig.json#L16

Reply:
Yup, that's it. I removed all references to `toastr` from my local `global.d.ts`, and the auto complete worked.

![image](https://github.com/SillyTavern/SillyTavern/assets/9962104/b3529c3f-700f-497b-ae39-10c66479293b)

Huh. So seems like it makes sense to import all the packages that are actually easily importable if they have a type definition, and then add docs for the others that aren't?

Because I am quite sure a few of those ones I added still make sense. Unless there are actual jQuery docs you can import for select2 and stuff.  
I don't know enough about how code docs in JS work, so maybe you know more than I do. I just know that none of the ones I changed here had any auto-complete or actual type checking for me in VS Code.

Reply:
I don't know which of the types in jsconfig actually work. It requires some trial and error, but I'm afraid most of them don't.

Reply:
I can say all of the ones I added here did not have any type checking or documentation yet. Doesn't mean they *can't* have those if they would've been imported cleanly or anything though, dunno.

Seems like for toastr the issue was that the global definition did override the jsconfig type import.  

I can try to check the other types whether they have an actual useful import or something over the weekend. But in the end, I'd be happy to see if we start to provide actual type-checking, that the libs we use support type-checking too :D

Lemme know if I should work through some stuff for this PR for before it can be merged.